src/user_agent.js

```diff
@@ -0,0 +1,27 @@
+module.exports = () => {
+  if ( typeof window === 'undefined' )
+    return 'nodejs';
+
+  // IE11 doesn't have window.navigator, test differently
+  // https://stackoverflow.com/questions/21825157/internet-explorer-11-detection
+  if (!!window.MSInputMethodContext && !!document.documentMode)
+    return 'Desktop';
+
+  if (/Android/i.test(window.navigator.userAgent)) {
+    if (/Firefox/i.test(window.navigator.userAgent))
+      return 'Android-Firefox';
```

Comment:
Not entireley, because the line to distinguish IE (above at line 7) should be still there, since certain IE versions do not have support for the `window.navigator`. I don't exactly know how IE behaves when we try to access the `window.navigator`, so I did not want to change this. Then I'd first have to try out whether this works on IE, and that would cost some time. That didn't seem very relevant time spending to me.

Of course I could have removed the lines below line 12, since from then we know for sure that whatever the outcome will be, it wil not be Android-Firefox anymore. I however thought, let's keep it consistent with `irma-frontend`, if we then have to update this logic we don't have to look for all the user agent code all around.

In `irma-frontend` the user agent logic is not exported publicly, and I don't want to secretly use this by circumventing this. In `irma-frontend` there is also no detection for Firefox now, so then I would first have to update `irma-frontend` for a change that is not even relevant for `irma-frontend`, but only because I want to use the private functions somewhere outside. This is not very solid.